### ISSUE TITLE
Show gpg signature in main view

### ISSUE THREAD

**Issue by varac** [jonas/tig#754]
I'm working with a few git repos that gpg sign their commits by default.
Signatures are shown in the diff view, but it would be awesome to show them in main view as well (i.e. by some nice UTF-8 icons like `⚷` or `✍` - or even better ones that I don't know so far).

**Comment by jonas** [comment]
I agree this type of hint would be a great addition. Maybe this could be shown before the branch labels.

**Comment by varac** [comment]
I propose to show it before the commiters name so it indicates a proper signature by the commiter.
But since you can customize tig's layout, I'd be also fine with your proposal.

**Comment by qrevel** [comment]
Maybe the author column color can change whether the commit is sign or not (red is not signed, green is signed).
It avoids extra space for an other column.

**Comment by andreasstieger** [comment]
Note that there is no canonical way of retrieving the keys required to verify these signatures. Core git relies on the local GnuPG keyring. Likewise trust for keys is delegated to the GnuPG trust systems, other than, say, GitHub, where the trust for a key is established by having the user upload their public key to their account.

**Comment by varac** [comment]
I'd be perfectly fine with showing the local GnuPG keyring trust, everything else would be more work imo.
Meaning, the user should be responsible for retrieving the keys outside of tig.

**Comment by userWayneCampbell** [comment]
Any updates on this feature?

**Comment by andreasstieger** [comment]
Nope.

**Comment by jonas** [comment]
I haven't had time to finish up the WIP. This will slow down loading of the main view quite a lot because Git will run gnupg for each commit. I guess that is fine for a first implementation. Later, we could look into fetching the signatures only for the visible commits using an extra load phase.

**Comment by hossbeast** [comment]
Was this (or similar) feature ever implemented?

**Comment by maks-ucs** [comment]
@hossbeast seems that it was implemented in 261c2763dda60b0aa421ca801cc032b33650f68f but using tig 2.5.8 from homebrew setting `set main-view-author-signature = yes` in my `.tigrc` doesn't seem to work.

**Comment by jcharaoui** [comment]
@maks-ucs The commit you linked to is only part of the `commit-signatures` branch, not the `master` branch.

As such, it has not been actually integrated.

**Comment by hossbeast** [comment]
> This will slow down loading of the main view quite a lot because Git will run gnupg for each commit.

Could this be done much more cheaply by displaying only "signature exists" instead of verifying the signature?
